src/cache/events/channel.rs

```diff
@@ -32,14 +32,13 @@ impl UpdateCache for ChannelDelete {
 
         cache.guilds.alter(&guild_id, |_, mut guild| {
             guild.channels.remove(&self.id);
-            guild.active_thread_parents = guild
+
+            guild
                 .active_thread_parents
-                .into_iter()
-                .filter(|(_, channel_id)| channel_id != &self.id)
-                .collect();
+                .retain(|_, &mut channel_id| channel_id != self.id);
```

Comment:
Does this really need to be mutable?

Reply:
It isn't mutable, this is telling the compiler to expect a mutable reference as an argument and copy it's value. So the type of channel_id is `Id<ChannelMarker>`. This has to be done because `HashMap::retain` has signature `pub fn retain(&mut self, f: FnMut(&Key, &mut Value) -> bool) {}` - note that it's `&`*`mut`*`Value`.